plugin/discovery/meta.go

```diff
@@ -16,21 +14,14 @@ type PluginMeta struct {
 	Name string
 
 	// Version is the semver version of the plugin, expressed as a string
-	// that might not be semver-valid. (Call VersionObj to attempt to
-	// parse it and thus detect if it is invalid.)
-	Version string
+	// that might not be semver-valid.
+	Version VersionStr
```

Comment:
I don't think there's any reason to use VersionStr as the TypeSpec here, since itself is just a string. It makes me wonder if there is some intended consequence I'm missing.